integrations/observability/grafana/resources.yaml

```diff
@@ -0,0 +1,220 @@
+        endpoint: http://localhost:9090/api/v1/otlp
+        tls:
+          insecure: true
+      otlphttp/traces:
+        endpoint: http://localhost:4418
+        tls:
+          insecure: true
+      otlphttp/logs:
+        endpoint: http://localhost:3100/otlp
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp, zipkin]
+          processors: [batch]
+          exporters: [otlphttp/traces]
+        metrics:
+          receivers: [otlp,prometheus/collector]
+          processors: [batch]
+          exporters: [otlphttp/metrics]
+        logs:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [otlphttp/logs]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  namespace: observability
+  labels:
+    app: otel-collector
+spec:
+  ports:
+    - name: otlp-http
+      port: 4318
+    - name: metrics
+      port: 8888
+  selector:
+    app: otel-collector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-collector
+  namespace: observability
+  labels:
+    app: otel-collector
+spec:
+  selector:
+    matchLabels:
+      app: otel-collector
+  template:
+    metadata:
+      labels:
+        app: otel-collector
+    spec:
+      containers:
+        - name: otel-collector
+          image: otel/opentelemetry-collector:latest
+          command:
+            - "/otelcol"
+            - "--config=/conf/otel-collector-config.yaml"
+          ports:
+            - containerPort: 4318
+            - containerPort: 8888
+          volumeMounts:
+            - name: otel-collector-config-vol
+              mountPath: /conf
+      volumes:
+        - configMap:
+            name: otel-collector-conf
+            items:
+              - key: otel-collector-config
+                path: otel-collector-config.yaml
+          name: otel-collector-config-vol
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-collector-conf
+  namespace: observability
+  labels:
+    app: otel-collector
+data:
+  otel-collector-config: |
+    ---
+    receivers:
+      otlp:
+        protocols:
+          http:
+            endpoint: 0.0.0.0:4318
+      prometheus:
+        config:
+          scrape_configs:
+            - job_name: 'kubearchive-otel-collector'
+              static_configs:
+                - targets: ['localhost:8888']
+
+    processors:
+      batch:
+
+    exporters:
+      debug:
+      otlphttp:
+        endpoint: "http://grafana-lgtm.observability.svc.cluster.local:4318"
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [debug, otlphttp]
+        metrics:
+          receivers: [otlp, prometheus]
+          processors: [batch]
+          exporters: [debug, otlphttp]
+        logs:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [debug]
```

Comment:
Please consider using kustomize with a configmapgenerator for extracting this to an external file

Reply:
Right, it will be much better. Done.